### ISSUE TITLE
Strange behaviour when processing is temporarily slow

### ISSUE THREAD

**Issue by royyeah** [awslabs/amazon-kinesis-client#952]
As consumer it makes sense in my opinion to have some control over the ingestion rate, so we rate limit how much we are consuming. This can lead to strange behavior due to time-outs as in some scenario's it can take up to more than 35s to process a batch of records. There is not enough configuration to let it function properly beyond this time-out as some things to control this are hardcoded (see issues referred to below). We could get it below this threshold if we could control `maxRecords` per batch but this is not possible using enhanced fan-out (also mentioned in issues below).

 I saw several people having similar issues:

 * https://github.com/awslabs/amazon-kinesis-client/issues/653 
 * https://github.com/awslabs/amazon-kinesis-client/issues/569
 * https://github.com/awslabs/amazon-kinesis-client/issues/463

**Comment by royyeah** [comment]
@zengyu714 @sahilpalvia Sorry for addressing you guys personally in advance, but just wanted to grab some attention as nobody replied anything helpful to the previous issues I refer to. Can you guys help out?

**Comment by charles-wilt** [comment]
@royyeah did you ever find a solution?

**Comment by royyeah** [comment]
No, unfortunately notOp 8 nov. 2023 om 16:44 heeft charles-wilt ***@***.***> het volgende geschreven:﻿
@royyeah did you ever find a solution?
